### PR TITLE
Rehash fix

### PR DESCRIPTION
**Pull request by MarkGibbons**
# Description

Resolves the objections to #52 and merges #52

## Issues Resolved
Fixes PR #52 which has been sitting around too long.

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by MarkGibbons** [comment]
Fixes the objections to #52 and merges #52

File: kitchen.yml [hunk 1]
```diff
@@ -5,31 +5,23 @@ driver:
 
 provisioner:
   product_name: chef
-  product_version: <%= ENV['CHEF_VERSION'] || "current" %>
+  product_version: 15
```
Comment (Xorima) [inline kitchen.yml:8]: Why not "current" or 16?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -44,8 +44,7 @@ jobs:
           - 'centos-7'
           - 'oracle-linux'
           - 'amazon-linux'
-          - 'ubuntu-1604'
-          - 'ubuntu-1804'
```
Comment (ramereth) [inline .github/workflows/ci.yml:48]: I'd keep 18.04 at least. You might also add CentOS 8 but that might want to be a different PR depending how much work that might be.

File: kitchen.yml (hunk 1 above)
Comment (ramereth) [reply kitchen.yml:8]: I agree, let's keep this as it was.

Comment (MarkGibbons) [reply kitchen.yml:8]: I changed it because it didn't work.  I'll update it to use latest instead of current.  Thanks for the comments.

File: templates/pyenv.sh [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
     pyenv_root="<%= @global_prefix %>"
     export PYENV_ROOT=${pyenv_root}
+    # Rehashing will fail in a system install
+    pyenv_init="pyenv init - --no-rehash"
 fi
 
 if [ -n "$pyenv_root" ]; then
     export PATH="${pyenv_root}/bin:$PATH"
-    eval "$(pyenv init -)"
+    eval "$($pyenv_init)"
```
Comment (darwin67) [inline templates/pyenv.sh:14]: There's no `else` statement above so it won't do anything in that case. is that expected?

**Comment by darwin67** [comment]
could you rebase your changes onto master?
there are too much noise right now in the listed commits and it's going to look pretty weird when you try to merge it.

**Comment by ramereth** [comment]
> could you rebase your changes onto master?
> there are too much noise right now in the listed commits and it's going to look pretty weird when you try to merge it.

Yes I agree. I'll work on doing this later today.

File: templates/pyenv.sh (hunk 3 above)
Comment (MarkGibbons) [reply templates/pyenv.sh:14]: yes.  Do you have a suggested action?

Comment (MarkGibbons) [reply templates/pyenv.sh:14]: PR #52 was sitting there in approved status for a long time. I'm trying to get the change merged.  I really don't have any opinion about the action to be taken, I'm trying to clean up the cruft, things like tickets that just need a rebase before they get merged or that don't execute tests.  You'll have a much better feel for what should happen in this situation.  Thanks

**Comment by ramereth** [comment]
@MarkGibbons looks like your force push overwrote the changes I made a few days ago with the history on this branch. I'll rebase this again to fix it.

File: templates/pyenv.sh (hunk 3 above)
Comment (darwin67) [reply templates/pyenv.sh:14]: oh nevermind, it was like this before. please disregard my comment.
it's been quite a while since I wrote this so forgot about it.

File: test/integration/user_install/controls/user_install_spec.rb [hunk 4]
```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
```
Comment (darwin67) [inline test/integration/user_install/controls/user_install_spec.rb:1]: May I asked why are tests being duplicated here?
I see you've made changes to `/test/integration/user_isntall/controls/user_install.rb`, is that not enough?

File: test/integration/system_install/controls/system_install_spec.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
```
Comment (darwin67) [inline test/integration/system_install/controls/system_install_spec.rb:1]: Same here. Seems like tests are being duplicated.

File: test/integration/user_install/controls/user_install_spec.rb (hunk 4 above)
Comment (MarkGibbons) [reply test/integration/user_install/controls/user_install_spec.rb:1]: I tried to merge Lance's rebase into the version I was working.  Mucked things up good. Sorry, I'll fix things.

File: kitchen.dokken.yml [hunk 6]
```diff
@@ -14,32 +14,25 @@ verifier:
...
 
 platforms:
-  # - name: debian-8
-  #   driver:
-  #     image: dokken/debian-8
-  #     intermediate_instructions:
-  #       - RUN /usr/bin/apt-get update
-  #       - RUN /usr/bin/apt-get install sudo -y
-
   - name: debian-9
```
Comment (Xorima) [inline kitchen.dokken.yml:17]: Do you think we should add dev 10 in this change, or in a future one?

Comment (ramereth) [reply kitchen.dokken.yml:17]: If it's a simple add, then I say do it here. If it's a bit more, then do it in another PR.

**Comment by ramereth** [comment]
@MarkGibbons I'll wait until all the fixes are done and do a rebase since I seem to mess this up for you when I do the rebase
